### PR TITLE
Add temporary egghead.io backwards compatibility (resolve #120)

### LINKED ISSUE #120: Add temporary egghead.io backwards compatibility
Add “Upload” and “Edit” lesson options and “Queue” link so all work can be done from the new Instructor Center, until native implementations are done.

- Add a lesson uploading button with the new `${lesson.upload_lesson_http_url}?lesson=${lesson.slug}&returnTo=/lessons/${lesson.slug}`
- on egghead.io, make “Replace Video” button be “Upload Video” if no video exists for lesson `/edit`

Once this is done, the big red button on egghead.io can be removed and all lesson uploading can start from the instructor center.

Back end pieces are complete: https://github.com/eggheadio/egghead-rails/pull/1110

### PR DESCRIPTION
**Pull request by trevordmiller**
# Changes

- Add temporary backwards compatible edit/upload video button to the individual lesson view which links to egghead.io; so users can do everything for a lesson from the Instructor Center for now until these things are all built inline in the Instructor Center
- Add temporary  backwards compatible queue link to navigation; so users can view queue on egghead.io for now until this is built inline in the Instructor Center
- Clean up navigation items
- Move node_modules imports to top with other node_modules imports
- Only fetch logged in instructor data once

# Result For User

screen shot 2017-02-17 at 4 32 39 pm
screen shot 2017-02-17 at 4 32 22 pm

image

---

### CODE REVIEW

File: src/components/InstructorsOnly/index.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import React from 'react'
+import {Error} from 'egghead-ui'
 import {instructorsOnlyDescriptionText} from 'utils/text'
 import Main from 'components/Main'
-import {Error} from 'egghead-ui'
```
Comment (trevordmiller) [inline src/components/InstructorsOnly/index.js]: I moved all `egghead-ui` imports next to the other node_modules at the top of each file

Comment (trevordmiller) [inline src/components/InstructorsOnly/index.js:4]: I moved all egghead-ui imports next to the other node_modules at the top of each file

File: src/index.js [hunk 2]
```diff
@@ -41,117 +42,125 @@ const App = () => {
     <Request url={decodedToken.user_url}>
       {({data}) => {
 
-        if(data && !data.instructor_url) {
+        const user = data
```
Comment (trevordmiller) [inline src/index.js:45]: I renamed the request data to be more explicit and avoid confusion with inner requests

File: src/index.js [hunk 3]
```diff
@@ -41,117 +42,125 @@ const App = () => {
...
-                  />
-
-                  <Route component={RouteNotFound} />
-
-                </Switch>
-
-              </Main>
-
-            </div>
+            <Request url={data.instructor_url}>
```
Comment (trevordmiller) [inline src/index.js:57]: Now the logged in instructor data is only fetched once and passed down

File: src/index.js [hunk 4]
```diff
@@ -41,117 +42,125 @@ const App = () => {
...
+                          action: '/',
+                        },
+                        {
+                          text: newLessonsTitleText,
+                          action: '/lessons/new',
+                        },
+                        {
+                          text: queueTitleText,
+                          action: temporaryQueueUrl,
+                        },
```
Comment (trevordmiller) [inline src/index.js:78]: Adds a temporary "Queue" link to navigation that links to egghead.io, until we build a queue in the Instructor Center directly

File: src/screens/Lessons/screens/Lesson/components/LessonData/index.js [hunk 5]
```diff
@@ -56,25 +60,49 @@ export default ({lesson}) => {
...
       children: lesson.wistia_id
         ? <WistiaVideo wistiaId={lesson.wistia_id} />
-        : null
+        : isInstructorsOwnLesson 
+          ? <Anchor url={lesson.lesson_http_url}>
+              <Button type='primary'>
+                {uploadVideoTitleText}
+              </Button>
+            </Anchor>
+          : noVideoDescriptionText
```
Comment (trevordmiller) [inline src/screens/Lessons/screens/Lesson/components/LessonData/index.js]: Adds a temporary "Upload" button that links to the lesson edit page on egghead.io, until we build a queue in the Instructor Center directly

File: src/screens/Lessons/screens/Lesson/components/LessonData/index.js [hunk 6]
```diff
@@ -56,25 +60,49 @@ export default ({lesson}) => {
...
+                type='warning'
+                size='small'
+                outline
+              >
+                {editLessonTitleText}
+              </Button>
+            </Anchor>
           </div>
-        </div>
-      ))}
```
Comment (trevordmiller) [inline src/screens/Lessons/screens/Lesson/components/LessonData/index.js]: Adds a temporary "Edit" button that links to the lesson edit page on egghead.io, until we build a queue in the Instructor Center directly

File: src/screens/Lessons/screens/Lesson/components/LessonNextSteps/index.js [hunk 7]
```diff
@@ -12,7 +12,7 @@ export default ({lesson}) => (
         {statusTitleText}
       </Heading>
       <div className={`
-        mb3 ttu tc pv2 ph3 br2 ba dib
+        mb3 ttu tc pv2 ph3 br2 ba b--dashed dib
```
Comment (trevordmiller) [inline src/screens/Lessons/screens/Lesson/components/LessonNextSteps/index.js:15]: Changed styles of lesson status to not look like a button

**Review by tayiorbeii — APPROVED** [review]

**Pull request merged**